### DISCUSSION TITLE
bo.export_qonnx error

### DISCUSSION THREAD

**Discussion by smtalds** [Xilinx/finn#874] [Q&A] [answered]
I got error on #870. I have 2 pc and when i check branch they are same. But my docker is installed torch version 1.13. But in other pc  installed 1.7 (like on requirements.txt, when i checked ). Same notebook is working other pc. I guess i got error because of this. How can i change torch?  When i write to community i can not get answer noone.

![Screenshot from 2023-08-17 10-54-56](https://github.com/Xilinx/finn/assets/59387657/600f4678-03d9-4855-9ef5-a70d4711389a)

**Comment by hleblevec** [comment]
Hi @smtalds,

My guess is you are working with two different branches and/or commit versions. Can you make sure you have the FINN repository in the same "state" on both machines? 
Otherwise, if you need to change the torch version you can do it just like any pip package running `pip install --upgrade torch==$version_number$`, even within the docker environment.

**Reply by smtalds** [reply]
@fpjentzsch  is told me for #870 error , run quicktest on dev branch. It's worked. Now i change same branch and commit version like  other pc . I am running now quicktest , after that i will give feedback. Thanks for answer. 
Same branch and commit : 
![WhatsApp Image 2023-08-17 at 13 21 47](https://github.com/Xilinx/finn/assets/59387657/1562697c-58ea-4bf9-a25c-cd6b5b8fbd51)
![Screenshot from 2023-08-17 13-37-47](https://github.com/Xilinx/finn/assets/59387657/188e2dc0-64dd-4e8d-bdbe-43d22ca5a356)

**Reply by smtalds** [reply]
![Screenshot from 2023-08-17 13-48-29](https://github.com/Xilinx/finn/assets/59387657/4004f636-b9a7-4a33-a9c8-b02e40a22f56)
Yep , i got same error :/ . I dont understand anything. I am running only quicktest and notebook but its not working.

**Reply by smtalds** [reply]
@hleblevec  Hi again, and torch --upgrade is not working because of python version. Main is running 3.8 , dev is running 3.10.

**Comment by smtalds** [comment, marked as answer]
> I got error on #870. I have 2 pc and when i check branch they are same. But my docker is installed torch version 1.13. But in other pc installed 1.7 (like on requirements.txt, when i checked ). Same notebook is working other pc. I guess i got error because of this. How can i change torch? When i write to community i can not get answer noone.
> 
> ![Screenshot from 2023-08-17 10-54-56](https://user-images.githubusercontent.com/59387657/261238737-600f4678-03d9-4855-9ef5-a70d4711389a.png)

I found the answer on https://github.com/Xilinx/finn/issues/853#issuecomment-1640105304. You have to use ishape with torch.rand(ishape).

Maybe you can update for  notebooks/end2end_example/bnn-pynq/cnv_end2end_example.ipynb @hleblevec @auphelia
